packages/peregrine/src/hooks/useCarousel.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useCallback } from 'react';
+
+export const useCarousel = carouselSize => {
+    const [activeItemIndex, setActiveItemIndex] = useState(0);
+    const [currentImageLoaded, setCurrentImageLoaded] = useState(false);
```

Comment:
Now that I think about it I think this belongs in the Carousel as this is a `DOM` concern rather than a data concern. Still, we can "hookify" this there.